html/captive-portal/lib/captiveportal/PacketFence/Controller/TLSProfile.pm

```diff
@@ -48,12 +48,22 @@ sub index : Path : Args(0) {
     my $request = $c->request;
     my $mac = $c->portalSession->clientMac;
 
+    my $provisioner = $c->profile->findProvisioner($mac);
+    my $pki_provider = $provisioner->getPkiProvider();
+    my $pki_provider_name = ref($pki_provider);
+    $pki_provider_name =~ s#^.*:##;
+
```

Comment:
Finding the provisioner should be moved below the user validation to avoid unnecessary work.